src/main/java/sx/blah/discord/api/internal/DiscordClientImpl.java

```diff
@@ -151,9 +154,9 @@ private void changeAccountInfo(String username, String avatar) throws RateLimitE
 					new StringEntity(DiscordUtils.GSON.toJson(new AccountInfoChangeRequest(username, avatar))));
 			AccountInfoChangeResponse response = DiscordUtils.GSON.fromJson(json, AccountInfoChangeResponse.class);
 
-			if (!this.getToken().equals(response.token)) {
+			if (!this.getToken().equals("Bot " + response.token)) {
 				Discord4J.LOGGER.debug(LogMarkers.API, "Token changed, updating it.");
```

Comment:
Actually isn't, I check the local PREFIXED token against the remote token AFTER prefixing the remote one